cli/src/api/pcluster_api.py

```diff
@@ -104,9 +105,11 @@ def delete_cluster(cluster_name: str, region: str, keep_logs: bool = True):
                 os.environ["AWS_DEFAULT_REGION"] = region
             # retrieve cluster config and generate model
             cluster = Cluster(cluster_name)
-            PclusterApi._check_version_2(cluster)
             cluster.delete(keep_logs)
-            return FullClusterInfo(cluster)
+            if PclusterApi._is_version_2(cluster):
+                return ClusterInfo(cluster.stack)
+            else:
+                return FullClusterInfo(cluster)
```

Comment:
I created two different classes to return an object rather than a dictionary from the api but let's think to both of them like two dictionaries, on which clusterInfo contains a subset of keys.

The difference is that ClusterInfo only contains info from the Stack, so it can be build for cluster created with 2.x and it's the easiest way to return a minimal set of info when listing multiple clusters/stacks without the need to retrieve the config for all the clusters.

The FullCluster info contains both information related to the stack, cluster-config and eventually info from running instances, so it can be returned only when interacting with an already created cluster (describe_cluster, delete_cluster, ...).

Anyway it's a good point, this code is still work in progress.